database/postgres/save_data.py

```diff
@@ -0,0 +1,169 @@
+                    status = status.get('long')
+
+                country_entry = game.get('location').get('country') if 'location' in game else game.get('country')
+                if country_entry:
+                    country_id = save_country_and_get_id(country_entry, country_dal)
+
+
+                league_entry = game.get('league')
+                if league_entry:
+                    process_entity_leagues([league_entry], sport_id, country_dal, session)
+                    league_id = league_dal.get_league_by_name_and_sport_id(league_entry.get('name') or league_entry, sport_id).league_id
+
+
+                teams_data = game.get('teams', {})
+                if teams_data:
+                    team_away_data = teams_data.get('away', {}) or teams_data.get('visitors', {})
+                    team_home_data = teams_data.get('home', {})
+                    process_entity_teams([team_away_data, team_home_data], sport_id, country_dal, session)
+                    team_away_id = team_dal.get_team_by_name_and_sport_id(team_away_data.get('name'), sport_id).team_index_id
+                    team_home_id = team_dal.get_team_by_name_and_sport_id(team_home_data.get('name'), sport_id).team_index_id
+
+
+                scores_data = game.get('scores', {}) or game.get('goals', {})
+                if scores_data:
+                    score_away_data = scores_data.get('away') or scores_data.get('visitors')
+                    score_home_data = scores_data.get('home')
+                    if not isinstance(score_home_data, int) and isinstance(score_home_data, dict):
+                        score_home_data = score_home_data.get('points') or score_home_data.get('total')
+                    if not isinstance(score_away_data, int) and isinstance(score_away_data, dict):
+                        score_away_data = score_away_data.get('points') or score_away_data.get('total')
+
+                game_dto = GameDTO(league_id=league_id if league_entry else None,
+                                   sport_id=sport_id,
+                                   country_id=country_id if country_entry else None,
+                                   team_away_id=team_away_id or None,
+                                   team_home_id=team_home_id or None,
+                                   score_away_team=score_away_data or None,
+                                   score_home_team=score_home_data or None,
+                                   status=status,
+                                   time=time,
+                                   date=date,
+                                   api_id=api_id)
+                game_dto_list.append(game_dto)
+            game_dal.save_games(game_dto_list)
+
+
+    except Exception as e:
+        session.rollback()
+        print("data was screwed up")
+        print(f"error processing data: {e}")
+
+
+def save_country_and_get_id(country_entry, country_dal: CountryDAL) -> int:
```

Comment:
`country_dal.create_country` method should already fetch the generated ID (since you are invoking `session.refresh(entity)`)

If that holds true (I might be wrong. But I do believe there should be a way of immediate Id fetching upon creation) - we can simplify quite a lot of logic and also have this method removed (`save_country_and_get_id`)